PhysicsTools/PatUtils/python/tools/runMETCorrectionsAndUncertainties.py

```diff
@@ -631,14 +631,14 @@ def getCorrectedMET(self, process, metType, correctionLevel,produceIntermediateC
         #Txy parameter tuning
         if "Txy" in correctionLevel:
             self.tuneTxyParameters(process, corScheme, postfix)
-            getattr(process, "patPFMetTxyCorr"+postfix).srcPFlow = self._parameters["pfCandCollection"].value
+            getattr(process, "patPFMetTxyCorr"+postfix).srcPFlow = "puppiForMET" if self.getvalue("Puppi") else self._parameters["pfCandCollection"].value
```

Comment:
These updates are technically not a backport, because they do not exhist in the master version of this PR.
Do they only affect nanoAOD productions, or they could potentially affect also miniAOD? There are no changes in miniAOD outputs, and that's reassuring: but I can hardly follow back all the path to be sure about it.

(By the way, would you have limited your updates to the real ones, avoiding changing white spaces here and there, the review could have been easier...)

Reply:
Technically these lines correspond to backports of features implemented in the master in:
https://github.com/cms-sw/cmssw/pull/29254/files#diff-79860e2d37e9cc0867e9de61faab9b79e0283ebf986856b7a05f517e6d1d083a

Reply:
(just in the master puppiForMET was renames in puppoNoLep and weights instead of weighted-candidates are used as input to MET)

Reply:
@ahinzmann @alefisico please let me know if these changes imply any possible effect on miniAOD, and if not which are the switches which prevents so (I admit I can hardly dig into the configuration to figure it out myself)